lib/spack/spack/repo.py

```diff
@@ -1362,6 +1362,33 @@ def get_pkg_class(self, pkg_name):
         if not inspect.isclass(cls):
             tty.die("%s.%s is not a class" % (pkg_name, class_name))
 
+        new_cfg_settings = (
+            spack.config.get("packages").get(pkg_name, {}).get("package_attributes", {})
+        )
+        overidden_attrs = getattr(cls, "overidden_attrs", {})
+        for key, val in overidden_attrs.items():
+            setattr(cls, key, val)
+        # Keep track of every class attribute that is overidden by the config:
+        # if the config changes between calls to this method, we make sure to
+        # restore the original config values (in case the new config no longer
+        # sets attributes that it used to)
+        new_overidden_attrs = {}
+        for key, val in new_cfg_settings.items():
+            if hasattr(cls, key):
+                new_overidden_attrs[key] = getattr(cls, key)
+            if isinstance(val, list):
+                raise spack.config.ConfigError("Unsupported attribute value: list")
+            elif isinstance(val, dict):
+                raise spack.config.ConfigError("Unsupported attribute value: dict")
+            if key not in ["git", "url", "submodules"]:
+                # Infer type
+                val = yaml.load(val)
```

Comment:
Isn't the value already read in from yaml? This seems redundant.

Reply:
This is so that users can write `x: true` instead of 

```
x:
  type: boolean
  value: true
```

Reply:
To be clear, this ensures that `x: true` is converted such that `Package.x is True` (otherwise it would be `Package.x == "true"`)